src/services/production.ts

```diff
@@ -123,11 +147,64 @@ export const services: ServiceConfigurator<SupportedTokens> = ($) => [
   // Routes
   [$.routes, {
     service: routes,
+    arguments: [
+      $.zoneRoutes,
+      $.meshRoutes,
+      $.wizardRoutes,
+      $.onboardingRoutes,
+      $.diagnosticsRoutes,
+    ],
+  }],
+
+  [$.meshRoutes, {
+    service: meshRoutes,
+    arguments: [
+      $.serviceRoutes,
+      $.gatewayRoutes,
+      $.dataplaneRoutes,
+      $.policyRoutes,
+    ],
+  }],
+
+  [$.dataplaneRoutes, {
+    service: dataplaneRoutes,
+  }],
+  [$.gatewayRoutes, {
+    service: gatewayRoutes,
+  }],
+  [$.serviceRoutes, {
+    service: serviceRoutes,
+  }],
+  [$.policyRoutes, {
+    service: policyRoutes,
     arguments: [
       $.store,
       $.Env,
     ],
   }],
+
+  [$.zoneRoutes, {
+    service: zoneRoutes,
+    arguments: [
+      constant(
+        [...(import.meta.env.VITE_ZONE_CREATION_FLOW === 'enabled' ? zoneActionRoutes() : [])],
+        {
+          description: 'kuma.zone.action.routes',
+        },
+      ),
+    ],
+  }],
+
+  [$.wizardRoutes, {
+    service: wizardRoutes,
+  }],
+  [$.onboardingRoutes, {
+    service: onboardingRoutes,
+  }],
+  [$.diagnosticsRoutes, {
+    service: diagnosticsRoutes,
+  }],
+
```

Comment:
A lot of this could be avoided if we make something to walk the routing config and merge it as if it was a tree of routes.